### PR TITLE
balance for users other than self

### PR DESCRIPTION
**Pull request by jackharrhy**
not tested because after spending 10 minutes trying to figure out how to run this bot + HyperionCS, i gave up because the /docs route didn't show how to create a new interaction :man_dancing:

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -174,20 +174,24 @@ async def daily(interaction: discord.Interaction):
...
             f"Error getting account.\n```\n{resp.json()['detail']}\n```",
         )
         return
 
+    prefix = "You" if user == interaction.user else f"{user.name}'s"
+
     account = Account(**resp.json())
     await interaction.response.send_message(
-        f"You have {account.balance} {currency.plural_form}."
+        f"{prefix} {account.balance} {currency.plural_form}."
```
Comment (nint8835) [inline bot.py:194]: This will currently give a format of something like
> You 20 Gumplets.

or

> Dan's 20 Gumplets.

File: bot.py [hunk 2]
```diff
@@ -174,20 +174,24 @@ async def daily(interaction: discord.Interaction):
 
 
 @tree.command(guild=guild)
-async def balance(interaction: discord.Interaction):
-    """Get your balance."""
-    resp = hyperion_session.get(
-        f"{config.hyperion_endpoint}/api/v1/accounts/{interaction.user.id}"
-    )
+async def balance(interaction: discord.Interaction, user: Optional[discord.user]):
```
Comment (nint8835) [inline bot.py]:
```
Traceback (most recent call last):
  File "/Users/nint8835/Development/hyperion-test-bot/bot.py", line 177, in <module>
    async def balance(interaction: discord.Interaction, user: Optional[discord.user]):
  File "/Users/nint8835/.pyenv/versions/3.10.0/lib/python3.10/typing.py", line 309, in inner
    return func(*args, **kwds)
  File "/Users/nint8835/.pyenv/versions/3.10.0/lib/python3.10/typing.py", line 400, in __getitem__
    return self._getitem(self, parameters)
  File "/Users/nint8835/.pyenv/versions/3.10.0/lib/python3.10/typing.py", line 525, in Optional
    arg = _type_check(parameters, f"{self} requires a single type.")
  File "/Users/nint8835/.pyenv/versions/3.10.0/lib/python3.10/typing.py", line 173, in _type_check
    raise TypeError(f"{msg} Got {arg!r:.100}.")
TypeError: typing.Optional requires a single type. Got <module 'discord.user' from '/Users/nint8835/.pyenv/versions/hyperion-test-bot/lib/python3.10/site-p.
```

**Review by nint8835 — CHANGES_REQUESTED** [review]

File: bot.py (hunk 2 above)
Comment (jackharrhy) [reply bot.py]: maintainer commits welcome

File: bot.py (hunk 1 above)
Comment (jackharrhy) [reply bot.py:194]: maintainer commits welcome

File: bot.py (hunk 2 above)
Comment (nint8835) [reply bot.py]:
```suggestion
async def balance(interaction: discord.Interaction, user: Optional[discord.User]):
```

File: bot.py [hunk 3]
```diff
@@ -174,20 +174,24 @@ async def daily(interaction: discord.Interaction):
...
+        user = interaction.user
+
+    resp = hyperion_session.get(f"{config.hyperion_endpoint}/api/v1/accounts/{user.id}")
     if resp.status_code != 200:
         await interaction.response.send_message(
             f"Error getting account.\n```\n{resp.json()['detail']}\n```",
         )
         return
 
+    prefix = "You" if user == interaction.user else f"{user.name}'s"
```
Comment (nint8835) [inline bot.py]:
```suggestion
    prefix = "You have" if user == interaction.user else f"{user.name} has"
```

**Review by nint8835 — CHANGES_REQUESTED** [review]

**Review by nint8835 — APPROVED** [review]

**Pull request merged**